cloudinit/net/__init__.py

```diff
@@ -572,16 +570,16 @@ def generate_fallback_config(config_driver=None):
         match = {
             "macaddress": read_sys_net_safe(target_name, "address").lower()
         }
+    if config_driver:
```

Comment:
This changes behavior, correct? Was the previous piece just wrong?

Reply:
> This changes behavior, correct? Was the previous piece just wrong?

No this behaves the same. It just defines `match` prior to initializing `cfg` - which allows mypy to reason about the type of `cfg` more correctly.

Reply:
Ah sorry, I didn't see the line after the `if driver` had to changed too.